improver/blending/weighted_blend.py

```diff
@@ -902,8 +920,9 @@ def process(self, cube, weights=None, cycletime=None, attributes_dict=None):
             result = self.weighted_mean(cube, weights)
         self._update_blended_metadata(result, attributes_dict)
 
-        # Re-mask output
-        if isinstance(cube.data, np.ma.core.MaskedArray):
-            result.data = np.ma.array(result.data)
+        # Checks the coordinate dimensions match the first relevant cube in the unblended cubeList.
+        result = check_cube_coordinates(
```

Comment:
I don't think any of the process unit tests use an input cube with a dimension with a single point in. Do you think we should add one so we don't get confused about this again?

Reply:
I think I've added one for process in test_TriangularWeightedBlendingAccrossAdjacentPoints that has a dimension with a single point, but I'll add one in for test_WeightedBlendAcrossWholeDimension.py too just to be sure